src/main/assisted-installer-controller/assisted_installer_main.go

```diff
@@ -304,3 +284,63 @@ func didInstallationFinish(clusterStatus string, log logrus.FieldLogger, status
 	}
 	return false
 }
+
+func getClusterAndDetermineTaintRemoval(ctx context.Context, assistedController assistedinstallercontroller.Controller,
+	ic inventory_client.InventoryClient, kc k8s_client.K8SClient, logger logrus.FieldLogger, invoker string) (*models.Cluster, bool) {
+	var cluster *models.Cluster
+	removeUninitializedTaint := false
+	if invoker == common.InvokerAgent {
+		if Options.ControllerConfig.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+			// When the agent-based installer installs a SNO cluster, assisted-service
+			// will never be reachable because it will not be running after the boostrap
+			// node reboots to become the SNO cluster. SetReadyState will never be
+			// able to get a connection to assisted-service in this circumstance.
+			//
+			// Instead of exiting with panic with "invalid memory address or nil pointer
+			// dereference" and having the controller restart because cluster is nil,
+			// log warning and exit 0. Otherwise the controller will keep restarting
+			// leaving the controller in a running state, even through the cluster
+			// has finished install.
+			logger.Warnf("cluster is SNO and invoker = %v, skipping assisted-installer-controller", invoker)
+			os.Exit(0)
```

Comment:
exit? not sure this is the right place

Reply:
I'm going to revert this function back into main and just return instead of calling os.Exit(0).